pkg/logs/sds/scanner.go

```diff
@@ -225,28 +225,50 @@ func interpretRCRule(userRule RuleConfig, standardRule StandardRuleConfig) (sds.
 		extraConfig.ProximityKeywords = sds.CreateProximityKeywordsConfig(userRule.IncludedKeywords.CharacterCount, userRule.IncludedKeywords.Keywords, nil)
 	}
 
-	// the RC schema supports multiple of them,
-	// for now though, the lib only supports one, so we'll just use the first one.
-	if len(standardRule.SecondaryValidators) > 0 {
-		received := standardRule.SecondaryValidators[0]
-		switch received.Type {
-		case RCSecondaryValidationChineseIdChecksum:
-			extraConfig.SecondaryValidator = sds.ChineseIdChecksum
-		case RCSecondaryValidationLuhnChecksum:
-			extraConfig.SecondaryValidator = sds.LuhnChecksum
-		default:
-			log.Warnf("Unknown secondary validator: ", string(received.Type))
-			// TODO(remy): telemetry
+	var defToUse = StandardRuleDefinition{Version: -1}
+
+	// go through all received definitions, use the most recent supported one.
+	// O(n) number of definitions in the rule.
+	for _, stdRuleDef := range standardRule.Definitions {
+		if defToUse.Version > stdRuleDef.Version {
+			continue
 		}
+
+		// The RC schema supports multiple of them,
+		// for now though, the lib only supports one, so we'll just use the first one.
+		if len(stdRuleDef.SecondaryValidators) > 0 {
```

Comment:
Would it make sense to `panic` if we see more than 1? Would the behavior of just choosing the first one be correct in that situation or would there be a silent error?

Reply:
It's a good point.

The RC schema supports sending multiples of them.
The SDS shared library does not support several of them.

There is probably a scenario where the shared library will support it in the future: the schema won't need any change, but the Agent will need an upgrade.

I don't think a `panic` is worth it: running only one of the secondary validators (among the eventual "many" in the feature) just means the amount of false positive would be higher (i.e. more redaction than necessary), which is OK from a security standpoint.

However, I'll add a warning log, and could eventually send a metric point for this (tag `too_many_validators` or similar). Does this work for you?

Reply:
That seems like a reasonable compromise. One question I have, though, is how difficult it is to just support it as-if there could be multiple? If it is trivial we could just do that now even though there will only be one in practice.

Reply:
We can't do that now and it's on the SDS team (it's the real SDS lib itself which doesn't support it for now).

Reply:
Gotcha. Given the impact is overeager redaction, as you mentioned, I'm ok with the warning.